lib/src/platform/src/codec.dart

```diff
@@ -645,6 +662,104 @@ class Codec extends StandardMessageCodec {
     );
   }
 
+  DeviceDetails _decodeDeviceDetails(Map<String, dynamic> jsonMap) {
+    FormFactor formFactor = _decodeFormFactor(
+        _readFromJson<String>(jsonMap, TxDeviceDetails.formFactor));
+
+    return DeviceDetails(
+      jsonMap[TxDeviceDetails.id] as String?,
+      jsonMap[TxDeviceDetails.clientId] as String?,
+      _decodeDevicePlatform(jsonMap[TxDeviceDetails.platform] as String),
+      formFactor,
+      toTypedJsonMap<String>(
+          jsonMap[TxDeviceDetails.metadata] as Map<Object?, Object?>?),
+      _decodeDevicePushDetails(
+        Map<String, dynamic>.from(jsonMap[TxDeviceDetails.devicePushDetails]
+            as Map<Object?, Object?>),
+      ),
+    );
+  }
+
+  DevicePushDetails _decodeDevicePushDetails(Map<String, dynamic> jsonMap) {
+    final jsonMapErrorReason =
+        jsonMap[TxDevicePushDetails.errorReason] as Map<Object?, Object?>?;
+
+    return DevicePushDetails(
+        Map<String, String>.from(
+            jsonMap[TxDevicePushDetails.recipient] as Map<Object?, Object?>),
+        _decodeDevicePushState(jsonMap[TxDevicePushDetails.state] as String?),
+        (jsonMapErrorReason != null)
+            ? _decodeErrorInfo(Map<String, dynamic>.from(jsonMapErrorReason))
+            : null);
+  }
+
+  LocalDevice _decodeLocalDevice(Map<String, dynamic> jsonMap) => LocalDevice(
+      _decodeDeviceDetails(jsonMap),
+      jsonMap[TxLocalDevice.deviceSecret] as String?,
+      jsonMap[TxLocalDevice.deviceIdentityToken] as String?);
+
+  FormFactor _decodeFormFactor(String? enumValue) {
+    switch (enumValue) {
+      case TxFormFactorEnum.phone:
+        return FormFactor.phone;
+      case TxFormFactorEnum.tablet:
+        return FormFactor.tablet;
+      case TxFormFactorEnum.desktop:
+        return FormFactor.desktop;
+      case TxFormFactorEnum.tv:
+        return FormFactor.tv;
+      case TxFormFactorEnum.watch:
+        return FormFactor.watch;
+      case TxFormFactorEnum.car:
+        return FormFactor.car;
+      case TxFormFactorEnum.embedded:
+        return FormFactor.embedded;
+      case TxFormFactorEnum.other:
+        return FormFactor.other;
+    }
+    throw AblyException(
+      'Platform communication error. FormFactor is invalid: $enumValue',
+    );
+  }
+
+  DevicePushState? _decodeDevicePushState(String? enumValue) {
+    switch (enumValue) {
+      case TxDevicePushStateEnum.active:
+        return DevicePushState.active;
+      case TxDevicePushStateEnum.failing:
+        return DevicePushState.failing;
+      case TxDevicePushStateEnum.failed:
+        return DevicePushState.failed;
+      default:
+        return null;
+    }
+    // throw AblyException(
+    //   'Platform communication error. DevicePushState is invalid: $enumValue',
+    // );
```

Comment:
It should never be null, but it is (there is a bug in ably-cocoa/ ably-java, actually they just forgot to implement that field completely 🐛 ). See https://github.com/ably/ably-flutter/issues/146

I've removed the comments since it can be null. 768d1f3b44e3d4e0fe0d02c3a3e99ec96f680bdb